aspnetcore/fundamentals/servers/kestrel/diagnostics.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Logging and diagnostics in Kestrel
+author: shirhatti
+description: Learn how to gather diagnostics from Kestrel.
+monikerRange: '>= aspnetcore-6.0'
+ms.author: soshir
+ms.date: 07/01/2021
+uid: kestrel/diagnostics
+---
+
+# Diagnostics in Kestrel
+
+By [Sourabh Shirhatti](https://twitter.com/sshirhatti)
+
+This article provides guidance for gathering diagnostics from Kestrel to help troubleshoot issues. Topics covered include:
+
+* **Logging** - Structured logs written to [.NET Core logging](xref:fundamentals/logging/index). <xref:Microsoft.Extensions.Logging.ILogger> is used by app frameworks to write logs, and by users for their own logging in an app.
+* **Metrics** - Representation of data measures over intervals of time, for example, requests per second. Metrics are emitted using `EventCounter` and can be observed using [dotnet-counters](/dotnet/core/diagnostics/dotnet-counters) command line tool or with [Application Insights](/azure/azure-monitor/app/eventcounters).
+* **DiagnosticSource** - DiagnosticsSource is a mechanism for production-time logging for rich data payloads for consumption within the process. Unlike logging, which assumes data will leave the process and expects serializable data, DiagnosticSource works well with complex data.
+
+## Logging
+
+Like most components in ASP.NET Core, Kestrel uses `Microsoft.Extensions.Logging` to emit log information. Kestrel employs the use of multiple [categories](xref:fundamentals/logging#log-category-1) which allows you to be selective on which logs you listen to.
```

Comment:
```suggestion
Like most components in ASP.NET Core, Kestrel uses `Microsoft.Extensions.Logging` to emit log information. Kestrel employs the use of multiple [categories](xref:fundamentals/logging/index#log-category-1) which allows you to be selective on which logs you listen to.
```

Reply:
@Rick-Anderson  Does the anchor tag not work on xref links? I want to link to a section of the doc.